hypervisor/dm/vpci/pci_pt.c

```diff
@@ -247,6 +247,10 @@ static void init_bars(struct pci_vdev *vdev, bool is_sriov_bar)
 	}
 	pbdf.value = vdev->pdev->bdf.value;
 
+	for (offset = 0U; offset < PCI_CFG_HEADER_LENGTH; offset += 4U) {
+		pci_vdev_write_cfg(vdev, offset, 4U, pci_pdev_read_cfg(pbdf, offset, 4U));
+	}
+
```

Comment:
for a VF, do we want to copy this kind of header? I think  optionally. 
Only when a VF is assigned to a VM, we need to copy. If a VF resides together with PF, it is not.
Please check with liuyuan.

Reply:
Ok

Reply:
Hi Fei & Eddie

If VF is not assigned, the SRIOV driver needs to access VF configuration space to check if VF has been created, the Linux SOS kernel will access "Reversion ID", "Class Code" , "Subsystem Vendor ID" and "Subsystem ID" of the first VF configuration space.

Another issue is do we need to copy all not assigned VFs configuration space,  from PCIe r4.0, sec 9.3.4.1, no require all VFs have some Reversion ID and Subsystem ID, but linux kernel currently assume they do, so I think we can follow the spec, copy all VFs configuration space is ok.